### ISSUE TITLE
Parse Utility/Commands Output

### ISSUE THREAD

**Issue by hooman96** [JustinJiaDev/SwiftShell#11]
parsing the outputs and provide useful options with commands for developers.

Bash : `ps | grep PID` 
SwiftShell : `ps(.pid)`

`ls -l | grep '^d'` = `ls(.dir)`

**Comment by JustinJiaDev** [comment]
The goal of SwiftShell is to use function calls to replace pipe. For example:

`grep("PID", from: ps())`

Or even variables:

```
let files = ls(.longFormat)
grep("^d", from: files)
```

The goal of SwiftShell is not to rewrite an existing shell using Swift, but to experiment with new ideas.

**Comment by hooman96** [comment]
Yes. it is good idea to use functions calls. But in this example:

`let files = ls(.longFormat)
grep("^d", from: files)`

Imagine, we have new developers who don't know how to use grep with "^d" or more complicated examples. We can provide informations as extra options: 

`ls(.dir)`

**Comment by JustinJiaDev** [comment]
Good point! But it should be handled by the ls command (or implement another command), not SwiftShell. Otherwise, there are way too many arguments to add and we cannot add all of them.

**Comment by hooman96** [comment]
I see. It is true that there are many arguments but I was trying to choose most useful cases (maybe by asking our users which arguments they want for each command)

**Comment by JustinJiaDev** [comment]
You can always write another command and pipe it yourself. We want to automate the process (if possible). Otherwise there are too many commands we can't implement all of them and add custom options.
